### PR TITLE
feat : skip tables

### LINKED ISSUE #18: Feature: skip table synchronization
Hi, a user asked me if it was possible to skip one or more tables from the synchronization. The use case was: some tables are pretty heavy (> 100GB) and do not need to be synchronized for a development environment. 

To skip the `public.logs` table, the configuration YAML could be like this:

```
source:
  connection_uri: postgres://root:password@localhost:5432/root
  skip:
    - database: public
      table: logs
  transformers:
    - database: public
      table: employees
      columns:
        - name: fist_name
          transformer: first-name
        - name: last_name
          transformer: random
bridge:
  bucket: replibyte-test
  region: us-east-2
  access_key_id: $AWS_ACCESS_KEY_ID
  secret_access_key: $AWS_SECRET_ACCESS_KEY
```

Feel free to pick this issue.

### PR DESCRIPTION
**Pull request by fabriceclementz**
Hi @evoxmusic 

I'm trying to resolve #18. At first, I would have adding `--exclude-table=` parameters according to this documentation https://www.postgresql.org/docs/9.3/app-pgdump.html  but It seems we can't with `pg_dumpall` as the name suggests.
So I'm trying to exclude rows during the stream reading.

Let me now, if you would have implemented this way.

----

- [x] Handle other cases like ALTER TABLE, COPY ...
- [x] Needs tests

### CODE REVIEW

File: replibyte/src/main.rs [hunk 1]
```diff
@@ -159,6 +159,11 @@ fn main() -> anyhow::Result<()> {
                         })
                         .collect::<Vec<_>>();
 
+                    let options = SourceOptions {
+                        transformers: &transformers,
+                        skip_config: source.skip.clone(),
```
Comment (evoxmusic) [inline replibyte/src/main.rs:164]: Maybe using a reference here?

File: replibyte/src/source/mod.rs [hunk 2]
```diff
@@ -10,7 +11,12 @@ pub mod postgres_stdin;
...
         &self,
-        transformers: &Vec<Box<dyn Transformer + '_>>,
+        options: SourceOptions,
         query_callback: F,
     ) -> Result<(), Error>;
 }
+
+pub struct SourceOptions<'a> {
+    pub transformers: &'a Vec<Box<dyn Transformer>>,
+    pub skip_config: Option<Vec<SkipConfig>>,
```
Comment (evoxmusic) [inline replibyte/src/source/mod.rs:21]: I would remove the `Option<Vec<..>>` and just keep the `Vec<...>` even if we have to create an empty vector

**Review by evoxmusic — CHANGES_REQUESTED** [review]
Nice one. However, can you write one or two tests to validate that it works as you do expect?

File: replibyte/src/source/mod.rs (hunk 2 above)
Comment (fabriceclementz) [reply replibyte/src/source/mod.rs]: 👍

File: replibyte/src/main.rs (hunk 1 above)
Comment (fabriceclementz) [reply replibyte/src/main.rs]: Done !

**Comment by fabriceclementz** [comment]
Hi @evoxmusic 
I think this pull request is ready to merge, let me know if you'd like that I make some changes :)

**Review by evoxmusic — APPROVED** [review]
Good job @fabriceclementz 🦾

**Pull request merged**